server/server.go

```diff
@@ -315,9 +315,13 @@ func (s *Server) CurrentState() (map[string]interface{}, error) {
 
 	totalQueued := 0
 	totalQueues := 0
+	// this map exposes all queues: backlog information
+	queues := make(map[string]interface{})
 	// queue size is cached so this should be very efficient.
```

Comment:
Unfortunately this comment is no longer accurate (it's RocksDB legacy).  We need to use a pipelined operation to Redis to avoid N network round trips to fetch the size of each queue in turn.  Here's how Sidekiq does it:

https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/api.rb#L147-L151

See storage/history.go for a Pipelined example.

Reply:
oh okay, i learnt something new today.
Happy :smiley:  to know we are using [redis: pipeline](https://redis.io/topics/pipelining).